src/components/sidenav/sidenav.js

```diff
@@ -244,16 +245,19 @@ function SidenavFocusDirective() {
  * @param {expression=} md-is-locked-open When this expression evaluates to true,
  * the sidenav 'locks open': it falls into the content's flow instead
  * of appearing over it. This overrides the `md-is-open` attribute.
+ * @param {expression=} md-is-locked-closed When this expression evaluates to true,
+ * the sidenav 'locks closed': hides sidenav.
```

Comment:
I'm still trying to get a handle on what behavior this API really has and how to name it.

Based on the naming, `md-is-locked-closed="true"` should make it so that you can't open the sidenav at all, even via `toggle()`. But in this [CodePen](https://codepen.io/Splaktar/pen/rorWzw) you can see that is not the case. This doesn't mean that your behavior is incorrect, it just means that we have the wrong naming of this API here.

Reply:
@Splaktar what if I change it to `md-is-hidden`, `is-hidden-when` or `md-hide-when`?
Basically by specifying the condition you determine when the sidenav has to become closed after it was opened.